src/deploy/functions/validate.ts

```diff
@@ -80,6 +82,25 @@ export function packageJsonIsValid(
   }
 }
 
+export function checkForInvalidChangeOfTrigger(
+  fn: CloudFunctionTrigger,
+  exFn: CloudFunctionTrigger
+) {
```

Comment:
I'd prefer throwing in place with a different error message for each case. That will help when customers come to support asking for questions (or can better explain an issue to a customer before they can reach out to support)